src/MultiSelect.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import { multiSelectedPropType, statusPropType } from './common-prop-types.js'
+import { Select } from './Select.js'
+import { Input } from './MultiSelect/Input.js'
+import { Menu } from './MultiSelect/Menu.js'
+import { FilterableMenu } from './MultiSelect/FilterableMenu.js'
+import { Loading } from './Select/Loading.js'
+
+const MultiSelect = ({
+    className,
+    selected,
+    tabIndex,
+    maxHeight,
+    onChange,
+    onFocus,
+    onBlur,
+    loading,
+    error,
+    warning,
+    valid,
+    disabled,
+    children,
+    clearable,
+    clearText,
+    filterable,
+    filterPlaceholder,
+    placeholder,
+    prefix,
+    empty,
+    loadingText,
+    noMatchText,
+    initialFocus,
+    dense,
+}) => (
+    <Select
+        className={className}
+        selected={selected}
+        input={
+            <Input
+                clearable={clearable}
+                clearText={clearText}
+                placeholder={placeholder}
+                prefix={prefix}
+            />
+        }
+        menu={
+            filterable ? (
+                <FilterableMenu
+                    empty={empty}
+                    noMatchText={noMatchText}
+                    placeholder={filterPlaceholder}
+                />
+            ) : (
+                <Menu empty={empty} />
+            )
+        }
```

Comment:
So there's the `Select`, the `SingleSelect` and the `MultiSelect`. The `Select` works as you're suggesting. It accepts a `menu` element and an `input` element. There you can just pass a component instance, and pass in a filterablemenu for example.

The `SingleSelect` and `MultiSelect` are basically convenience wrappers for the Select, for if you want a single or multi-select, without everything that a *Field component adds (the help-text, etc.).

I can imagine that a user of ui-core could encounter a situation where they need a simple select, so I thought it made sense to provide a component that can be used as a select without having to construct it yourself out of the various input and menu components. That seems cumbersome to have to do everytime, and I think would probably lead to a lot of app-local components that resemble what is currently the `SingleSelect` and `MultiSelect`. So I thought it would be good to provide that.

If they want more control, they can do as you're suggesting by using the `Select` component. And if they need the extras from the `*Field` components, they can use the `SingleSelectField` and `MultiSelectField` components.

Reply:
Ah, alright! Great that you thought about this already. Consider my comment rescinded.